PWGHF/D2H/Tasks/taskSigmacToCascade.cxx

```diff
@@ -0,0 +1,205 @@
+    registry.add("Data/hEtaSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #eta ; counts;", {HistType::kTH1F, {axisEta}});                                                                    /// Σc0
+    registry.add("Data/hEtaSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #eta ; counts;", {HistType::kTH1F, {axisEta}});                                                            /// Σc++
+    registry.add("Data/hEtaSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #eta ; counts;", {HistType::kTH1F, {axisEta}});                                                      /// Σc0,++
+    registry.add("Data/hPhiSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                                    /// Σc0
+    registry.add("Data/hPhiSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                            /// Σc++
+    registry.add("Data/hPhiSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                      /// Σc0,++
+
+    /// THn for candidate Λc+ and Σc0,++ cut variation
+    if (enableTHn) {
+      const AxisSpec thnAxisMassLambdaC{configAxisMassLambdaC, "inv. mass (K0s p) (GeV/#it{c}^{2})"};
+      const AxisSpec thnAxisPtLambdaC{thnConfigAxisPt, "#it{p}_{T}(#Lambda_{c}^{+}) (GeV/#it{c})"};
+      const AxisSpec thnAxisPtSigmaC{thnConfigAxisPt, "#it{p}_{T}(#Sigma_{c}^{0,++}) (GeV/#it{c})"};
+      const AxisSpec thnAxisDecLength{thnConfigAxisDecLength, "decay length #Lambda_{c}^{+} (cm)"};
+      const AxisSpec thnAxisDecLengthXY{thnConfigAxisDecLengthXY, "decay length XY #Lambda_{c}^{+} (cm)"};
+      const AxisSpec thnAxisCPA{thnConfigAxisCPA, "cosine of pointing angle #Lambda_{c}^{+}"};
+      const AxisSpec thnAxisCPAXY{thnConfigAxisCPAXY, "cosine of pointing angle XY #Lambda_{c}^{+}"};
+      registry.add("hnSigmaC0", "THn for Sigmac from Cascade channel", HistType::kTHnSparseF, {thnAxisMassLambdaC, thnAxisPtLambdaC, axisDeltaMassSigmaC, thnAxisDecLength, thnAxisDecLengthXY, thnAxisCPA, thnAxisCPAXY, thnAxisPtSigmaC});
+      registry.add("hnSigmaCPlusPlus", "THn for Sigmac from Cascade channel", HistType::kTHnSparseF, {thnAxisMassLambdaC, thnAxisPtLambdaC, axisDeltaMassSigmaC, thnAxisDecLength, thnAxisDecLengthXY, thnAxisCPA, thnAxisCPAXY, thnAxisPtSigmaC});
+      registry.add("hnSigmaC0PlusPlus", "THn for Sigmac from Cascade channel", HistType::kTHnSparseF, {thnAxisMassLambdaC, thnAxisPtLambdaC, axisDeltaMassSigmaC, thnAxisDecLength, thnAxisDecLengthXY, thnAxisCPA, thnAxisCPAXY, thnAxisPtSigmaC});
+    }
```

Comment:
is there any reason to have 3 `THnSparses` instead on 1 only that also includes an axis for the charge?

Reply:
Done.